crates/rooch-indexer/src/indexer_reader.rs

```diff
@@ -297,4 +299,64 @@ impl IndexerReader {
 
         Ok(result)
     }
+
+    pub fn sync_states(
+        &self,
+        cursor: Option<u64>,
+        limit: usize,
+        descending_order: bool,
+    ) -> IndexerResult<Vec<IndexerStateChangeSet>> {
+        let tx_order = if let Some(cursor) = cursor {
+            cursor as i64
+        } else if descending_order {
+            let max_tx_order: i64 = self.inner_indexer_reader.run_query(|conn| {
+                state_change_sets::dsl::state_change_sets
+                    .select(state_change_sets::tx_order)
+                    .order_by(state_change_sets::tx_order.desc())
+                    .first::<i64>(conn)
+            })?;
+            max_tx_order + 1
+        } else {
+            -1
+        };
+
+        let where_clause = if descending_order {
+            format!(" ({TX_ORDER_STR} < {})", tx_order)
+        } else {
+            format!(" ({TX_ORDER_STR} > {})", tx_order)
+        };
+        let order_clause = if descending_order {
+            format!("{TX_ORDER_STR} DESC")
+        } else {
+            format!("{TX_ORDER_STR} ASC")
+        };
+
+        let query = format!(
+            "
+                SELECT * FROM state_change_sets \
+                WHERE {} \
+                ORDER BY {} \
+                LIMIT {}
+            ",
+            where_clause, order_clause, limit,
+        );
+
+        tracing::debug!("sync states: {}", query);
+        let stored_state_change_sets = self
+            .inner_indexer_reader
+            .run_query(|conn| diesel::sql_query(query).load::<StoredStateChangeSet>(conn))?;
+
+        let result = stored_state_change_sets
```

Comment:
Maybe we should merge all ChangeSets into one ChangeSet to reduce the size in the future.